src/Compilers/Test/Utilities/CSharp/CSharpTestBase.cs

```diff
@@ -247,206 +293,125 @@ public static SyntaxTree ParseWithRoundTripCheck(string text, CSharpParseOptions
-        {
-            return CreateStandardCompilation(Parse(sources, parseOptions), references, options, assemblyName);
-        }
+            string sourceFileName = "",
+            string assemblyName = "") => CreateCompilation(Parse(source, sourceFileName, parseOptions), references, options, TargetFramework.Mscorlib46, assemblyName);
 
-        public static CSharpCompilation CreateStandardCompilation(
-            SyntaxTree syntaxTree,
+        public static CSharpCompilation CreateCompilationWithMscorlib46(
+            string[] source,
             IEnumerable<MetadataReference> references = null,
             CSharpCompilationOptions options = null,
-            string assemblyName = "")
-        {
-            return CreateStandardCompilation(new SyntaxTree[] { syntaxTree }, references, options, assemblyName);
-        }
-
-        private static readonly ImmutableArray<MetadataReference> s_stdRefs = CoreClrShim.IsRunningOnCoreClr
-            ? ImmutableArray.Create<MetadataReference>(NetStandard20.NetStandard, NetStandard20.MscorlibRef, NetStandard20.SystemRuntimeRef, NetStandard20.SystemDynamicRuntimeRef)
-            : ImmutableArray.Create(MscorlibRef);
-
-        // Careful! Make sure everything in s_desktopRefsToRemove is constructed with
-        // the same object identity, since MetadataReference uses reference equality.
-        // this may mean adding Interlocked calls in the construction of the reference.
-        private static readonly ImmutableArray<MetadataReference> s_desktopRefsToRemove = ImmutableArray.Create(SystemRef, SystemCoreRef);
+            CSharpParseOptions parseOptions = null,
+            string assemblyName = "") => CreateCompilation(source, references, options, parseOptions, TargetFramework.Mscorlib46, assemblyName);
 
-        public static CSharpCompilation CreateStandardCompilation(
-            IEnumerable<SyntaxTree> trees,
-            IEnumerable<MetadataReference> references = null,
+        public static CSharpCompilation CreateCompilationWithMscorlib45AndCSruntime(
+            string source,
             CSharpCompilationOptions options = null,
-            string assemblyName = "")
+            CSharpParseOptions parseOptions = null,
+            MetadataReference[] references = null)
         {
-            if (CoreClrShim.IsRunningOnCoreClr)
+            var refs = new List<MetadataReference>() { MscorlibRef_v4_0_30316_17626, SystemCoreRef, CSharpRef };
+
+            if (references != null)
             {
-                references = references?.Except(s_desktopRefsToRemove);
+                refs.AddRange(references);
             }
-            return CreateCompilation(trees, (references != null) ? s_stdRefs.Concat(references) : s_stdRefs, options, assemblyName);
+
+            return CreateCompilationWithNone(new[] { Parse(source, options: parseOptions) }, refs, options);
         }
 
         public static CSharpCompilation CreateCompilationWithMscorlibAndSystemCore(
```

Comment:
I find it strange that we still have an API that uses an unnamed "Mscorlib"

Reply:
This is fixed in #24962